### PR TITLE
Bring Language sample up to standard.

### PR DESCRIPTION
**Pull request by jmdobry**

### CODE REVIEW

File: language/README.md [hunk 1]
```diff
@@ -13,73 +13,50 @@ Learning API.
...
+__Usage:__ `node analyze --help`
+
+```
+Commands:
+  sentiment <text>                       Detect the sentiment of a block of text.
+  sentimentFromFile <bucket> <filename>  Detect the sentiment of text in a GCS file.
+  entities <text>                        Detect the entities of a block of text.
+  entitiesFromFile <bucket> <filename>   Detect the entities of text in a GCS file.
+  syntax <text>                          Detect the syntax of a block of text.
+  syntaxFromFile <bucket> <filename>     Detect the syntax of a block of text.
```
Comment (ace-n) [inline language/README.md]: Typo in description?

File: language/analyze.js [hunk 2]
```diff
@@ -12,163 +12,294 @@
...
+  };
 
-var languageScopes = ['https://www.googleapis.com/auth/cloud-platform'];
+  // See https://googlecloudplatform.github.io/google-cloud-node/#/docs/language/latest/language/document
+  document.detectSentiment(config, function (err, sentiment) {
+    if (err) {
+      return callback(err);
+    }
 
+    console.log('Found %s sentiment', sentiment.polarity >= 0 ? 'positive' : 'negative');
```
Comment (ace-n) [inline language/analyze.js]: Maybe include the actual polarity value?

File: language/analyze.js [hunk 3]
```diff
@@ -12,163 +12,294 @@
...
     if (err) {
       return callback(err);
     }
 
-    // Depending on the environment that provides the default credentials
-    // (e.g. Compute Engine, App Engine), the credentials retrieved may
-    // require you to specify the scopes you need explicitly.
-    if (authClient.createScopedRequired && authClient.createScopedRequired()) {
-      authClient = authClient.createScoped(languageScopes);
+    console.log('Found %s sentiment', sentiment.polarity >= 0 ? 'positive' : 'negative');
```
Comment (ace-n) [inline language/analyze.js]: Include actual polarity value?

File: language/analyze.js [hunk 4]
```diff
@@ -12,163 +12,294 @@
...
+// [END analyze_sentiment_from_file]
+
+// [START analyze_entities]
+/**
+ * Detect the entities from a block of text.
+ *
+ * @param {string} text The text to analyze.
+ * @param {object} [options] Optional configuration.
+ * @param {string} [options.language] The language of the text, e.g. "en".
+ * @param {string} [options.type] "text" or "html".
```
Comment (ace-n) [inline language/analyze.js]: Describe this param more? e.g. `The format of the text (either "text" or "html").`

File: language/analyze.js [hunk 5]
```diff
@@ -12,163 +12,294 @@
...
-}
+// [START analyze_entities_from_file]
+/**
+ * Detect the entities in a text file that resides in Google Cloud Storage.
+ *
+ * @param {string} bucket The bucket where the file resides.
+ * @param {string} filename The name of the file to be analyzed.
+ * @param {object} [options] Optional configuration.
+ * @param {string} [options.language] The language of the text, e.g. "en".
+ * @param {string} [options.type] "text" or "html".
```
Comment (ace-n) [inline language/analyze.js]: Ditto.

File: language/analyze.js [hunk 6]
```diff
@@ -12,163 +12,294 @@
...
+// [END analyze_entities_from_file]
+
+// [START analyze_syntax]
+/**
+ * Detect the syntax in a block of text.
+ *
+ * @param {string} text The text to analyze.
+ * @param {object} [options] Optional configuration.
+ * @param {string} [options.language] The language of the text, e.g. "en".
+ * @param {string} [options.type] "text" or "html".
```
Comment (ace-n) [inline language/analyze.js]: Ditto param description.

File: language/analyze.js [hunk 7]
```diff
@@ -12,163 +12,294 @@
...
+
+// [START analyze_syntax_from_file]
+/**
+ * Detect the syntax in a text file that resides in Google Cloud Storage.
+ *
+ * @param {string} bucket The bucket where the file resides.
+ * @param {string} filename The name of the file to be analyzed.
+ * @param {object} [options] Optional configuration.
+ * @param {string} [options.language] The language of the text, e.g. "en".
+ * @param {string} [options.type] "text" or "html".
```
Comment (ace-n) [inline language/analyze.js]: Ditto param description.

File: language/analyze.js [hunk 8]
```diff
@@ -12,163 +12,294 @@
...
+/**
+ * Detect the entities from a block of text.
+ *
+ * @param {string} text The text to analyze.
+ * @param {object} [options] Optional configuration.
+ * @param {string} [options.language] The language of the text, e.g. "en".
+ * @param {string} [options.type] "text" or "html".
+ * @param {function} callback The callback function.
+ */
+function analyzeEntities (text, options, callback) {
```
Comment (ace-n) [inline language/analyze.js]: Should this be something like `analyzeEntities_FromString_`?

File: language/analyze.js [hunk 9]
```diff
@@ -12,163 +12,294 @@
...
+
+cli
+  .demand(1)
+  .command('sentiment <text>', 'Detect the sentiment of a block of text.', {}, function (options) {
+    program.analyzeSentiment(options.text, utils.pick(options, ['language', 'type']), utils.makeHandler());
+  })
+  .command('sentimentFromFile <bucket> <filename>', 'Detect the sentiment of text in a GCS file.', {}, function (options) {
+    program.analyzeSentimentFromFile(options.bucket, options.filename, utils.pick(options, ['language', 'type']), utils.makeHandler());
+  })
+  .command('entities <text>', 'Detect the entities of a block of text.', {}, function (options) {
```
Comment (ace-n) [inline language/analyze.js]: I feel like these should be something like "entitiesFromText" or something like that. "Entities" gives me the impression that "entitiesFromText" is a conceptual subclass of this, while to me they appear to be on the same level conceptually.

Thoughts?

File: language/system-test/analyze.test.js [hunk 10]
```diff
@@ -13,50 +13,103 @@
...
+        assert.equal(typeof sentiment.magnitude, 'number');
+        done();
+      });
+    });
+  });
+
+  describe('analyzeEntities', function () {
+    it('should analyze entities in text', function (done) {
+      program.analyzeEntities(text, options, function (err, entities) {
+        assert.equal(null, err);
```
Comment (ace-n) [inline language/system-test/analyze.test.js]: Nit: I'm used to the constant being on the right for asserts, and that's the standard we seem to use elsewhere. Maybe a quick find-replace?

File: language/test/analyze.test.js [hunk 11]
```diff
@@ -13,6 +13,372 @@
...
+
+  describe('main', function () {
+    it('should call analyzeSentiment', function () {
+      var program = getSample().program;
+
+      sinon.stub(program, 'analyzeSentiment');
+      program.main(['sentiment', text]);
+      assert.equal(program.analyzeSentiment.calledOnce, true);
+      assert.deepEqual(program.analyzeSentiment.firstCall.args.slice(0, -1), [text, {
+        type: 'text',
```
Comment (ace-n) [inline language/test/analyze.test.js]: Nit: for brevity's sake, should we define `{ type: 'text', language: undefined }` as a var and reuse that?

**Comment by ace-n** [comment]
LGTM, aside from my comments.

File: language/analyze.js (hunk 2 above)
Comment (jmdobry) [inline language/analyze.js]: Both polarity and magnitude will get printed by the CLI. I don't think the number itself provides that much information beyond being positive or negative, at least not to me.

File: language/analyze.js (hunk 3 above)
Comment (jmdobry) [inline language/analyze.js]: re: https://github.com/GoogleCloudPlatform/nodejs-docs-samples/pull/190#discussion_r76510128

File: language/system-test/analyze.test.js (hunk 10 above)
Comment (jmdobry) [inline language/system-test/analyze.test.js]: I usually do that do, my bad.

File: language/test/analyze.test.js (hunk 11 above)
Comment (jmdobry) [inline language/test/analyze.test.js]: Sure

File: language/README.md (hunk 1 above)
Comment (jmdobry) [inline language/README.md]: Fixed.

File: language/analyze.js (hunk 4 above)
Comment (jmdobry) [inline language/analyze.js]: Fixed.

File: language/analyze.js (hunk 8 above)
Comment (jmdobry) [inline language/analyze.js]: Fixed.

File: language/analyze.js (hunk 9 above)
Comment (jmdobry) [inline language/analyze.js]: Fixed.

File: language/system-test/analyze.test.js (hunk 10 above)
Comment (jmdobry) [inline language/system-test/analyze.test.js]: Fixed.

**Comment by jmdobry** [comment]
@dpebot Merge when green.

**Pull request merged**